### PR TITLE
Apply changes to Shubhamrawat5 / spotify-playlist-downloader

### PR DESCRIPTION
**Pull request by b5i**
Hi !
Just added the metadata accuracy with album search and modify a class.

Can Participate to Hacktoberfest pls ?

### CODE REVIEW

**Comment by Shubhamrawat5** [comment]
You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

**Comment by b5i** [comment]
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Yes I'm working actually on it, the PR is not complete yet.

**Comment by b5i** [comment]
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

On apple music the update is done

**Comment by b5i** [comment]
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

For me it's working fine
Capture d’écran 2021-10-22 à 16 45 55
Just fix few line now for the lyrics

**Comment by b5i** [comment]
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Strange it's working one time on two...

**Comment by b5i** [comment]
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Ok I know, the program does not store on playlist-info.txt the album so it's undefined
I'm fixing that

**Comment by b5i** [comment]
Ok all is done and works now ! 💯

**Comment by b5i** [comment]
Capture d’écran 2021-10-22 à 17 02 56

**Comment by Shubhamrawat5** [comment]
so much changed! 

In spotify.js file, you've changed the query selector value to ".ax_46rB4EOK3hF__KEMO" , which seems pointless and breaking the code.

You put "ax_46rB4EOK3hF__KEMO" points to the number on side of song name
but previously it was "_qbBHRjaGvaZoEZDZ_IY" which point to the heading "3 likes 82 songs, 6 hr 7 min" , here total songs are mentioned so it was taking total song number from there

**Comment by b5i** [comment]
Ah yeah ok this thing, it was the only I doubt because first time I run the script it bugs and crash directly so I gladly change it.

**Comment by b5i** [comment]
I've just a question :
I had to download 'Breathe' and it downloads 'King' while it shouldn't download and with Apple Music it downloads the right track.
Is it a bug in the script or on the slider.kz side ?

Thanks

**Comment by Shubhamrawat5** [comment]
Not sure, but i've noticed one thing in slider.kz website that a single character or reorder of words change the whole search result.

If you just put a comma in search query, ordering of songs in result might change

**Comment by Shubhamrawat5** [comment]
Is the lyrics adding function complete?

**Comment by b5i** [comment]
> Is the lyrics adding function complete?

Yes, I have fixed all the bugs I faced. I use a lot the Apple Music one so I try to fix as much as possible all the bugs. I will try with a big playlist Spotify and see what's happening.
Apple Music : Fixed. Need to approve the 2 PRs. → I'll test a lot and improve this tool.
Spotify : Fixed too. Need to approve this PR. → I use less Spotify but I'll test it in few days.
Note : I forgot to add on README.md the parameters, I'll improve them tomorrow or day after tomorrow.
Note 2 : Roadmap : 
1. Complete parameters function (-p with playlist, and others things like this).
2. Try to put directly on the file the lyrics (no .lrc file).
3. Enjoy the program.
If you want to contact me without using GitHub my mail address is bollengier.antoine@icloud.com
Thanks for your help !

File: downloader.js [hunk 1]
```diff
@@ -5,18 +5,76 @@ const axios = require("axios");
...
+        console.log(`error: ${error.message}`);
+        lyricsok = false;
+        return;
+    }
+    if (stderr) {
+        console.log(`stderr: ${stderr}`);
+        lyricsok = false;
+        return;
+    }
+    if (`${stdout}` == 'LYRICS NOT FOUND\n'){
```
Comment (Shubhamrawat5) [inline downloader.js]: Here is a issue, i tried printing stdout value by JSON.stringify(stdout) and it gave 
"LYRICS NOT FOUND\r\n" 

so i think there is \r missing

**Comment by Shubhamrawat5** [comment]
Also Lyrics python code is not working properly, It didn't give a single lyrics for me.

I tried to check and found `for result_link in result_links:` is working fine but ` if artist.lower() in lower_title and songname.lower() in lower_title: ` this if condition is getting false everytime so lyrics are not generating.

If you fix this, please update in apple playlist repo too. That would be very helpful

**Comment by b5i** [comment]
Got this error : 
EXTRACTING TOTAL SONGS...
spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221
            throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
                  ^

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'innerText')
    at __puppeteer_evaluation_script__:4:9
    at ExecutionContext._evaluateInternal (/Users/toine/djo/spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async getPlaylistList (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:36:19)
    at async getPlaylistInfo (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:108:3)
    at async Object.module.exports.getPlaylist (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:126:3)

**Comment by Shubhamrawat5** [comment]
Try again, i think this happen when your network is slow.

**Comment by b5i** [comment]
> Try again, i think this happen when your network is slow.

I've try many many times but that just doesn't work...
I've reinstalled and do all those things but nothing to do...
My speed is 400MBs so I think it's enough to fetch a webpage

**Comment by b5i** [comment]
Very strange it worked yesterday.

**Comment by Shubhamrawat5** [comment]
wait, i know why that is happening

**Comment by Shubhamrawat5** [comment]
Put 
` await page.waitForSelector("div[aria-rowindex='2']");`

below `console.log("opened.");` line

**Comment by Shubhamrawat5** [comment]
That is happening because song data takes time to load after page get loaded, so `await page.waitForSelector("div[aria-rowindex='2']");` will hold the code till the first song is shown!

**Comment by b5i** [comment]
> That is happening because song data takes time to load after page get loaded, so `await page.waitForSelector("div[aria-rowindex='2']");` will hold the code till the first song is shown!

Sorry, it doesn't work, have you got another div or element that is generated in last time ?

File: downloader.js (hunk 1 above)
Comment (b5i) [reply downloader.js]: Capture d’écran 2021-10-23 à 20 32 50
Capture d’écran 2021-10-23 à 20 32 41
I didn't get this issue

**Comment by Shubhamrawat5** [comment]
It should work, div[aria-rowindex='2'] points to first song.

**Comment by b5i** [comment]
To be sure it's like this no ?
Capture d’écran 2021-10-24 à 10 53 29

**Comment by b5i** [comment]
> To be sure it's like this no ?
> Capture d’écran 2021-10-24 à 10 53 29

But I've got the same problem

**Comment by Shubhamrawat5** [comment]
Yes, weird it is working fine for me

**Comment by b5i** [comment]
> Yes, weird it is working fine for me

Very weird, which version of puppeteer have you got ?

**Comment by Shubhamrawat5** [comment]
"puppeteer": "^10.4.0",

Post the latest error again

**Comment by b5i** [comment]
EXTRACTING TOTAL SONGS...
spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221
            throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
                  ^

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'innerText')
    at __puppeteer_evaluation_script__:4:9
    at ExecutionContext._evaluateInternal (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async getPlaylistList (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:36:19)
    at async getPlaylistInfo (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:108:3)
    at async Object.module.exports.getPlaylist (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:126:3)

**Comment by b5i** [comment]
I've just removed a part of the path with my informations

**Comment by Shubhamrawat5** [comment]
Which playlist you're using?

**Comment by b5i** [comment]
> Which playlist you're using?

Check yours emails

**Comment by Shubhamrawat5** [comment]
Didn't get any mail yet. It's sr856161@gmail.com

**Comment by b5i** [comment]
> Didn't get any mail yet. It's [sr856161@gmail.com](mailto:sr856161@gmail.com)

Sent

**Comment by Shubhamrawat5** [comment]
Okay so Html is different for both playlist, so this is giving error

wait..

**Comment by b5i** [comment]
> Okay so Html is different for both playlist, so this is giving error
> 
> wait..

Maybe with the different location

**Comment by Shubhamrawat5** [comment]
Use this

` let totalSongsHtmlInfo = document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY")[
      document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY").length - 1
    ].innerText;`

**Comment by b5i** [comment]
> Use this
> 
> ` let totalSongsHtmlInfo = document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY")[ document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY").length - 1 ].innerText;`

It's better 
Ok now I get this, `throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));

Error: Evaluation failed: RangeError: Invalid array length`

**Comment by b5i** [comment]
Fixed !
Just add   ```await page.setExtraHTTPHeaders({
    'Accept-Language': 'en'
});``` at line 116

**Comment by Shubhamrawat5** [comment]
Pls make another fresh PR.

**Pull request closed without merging**